netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -106,20 +108,35 @@ public static ProtocolNegotiator serverPlaintext() {
 
   /**
    * Create a server TLS handler for HTTP/2 capable of using ALPN/NPN.
+   * @param executorPool a dedicated {@link Executor} pool for time-consuming TLS tasks
    */
-  public static ProtocolNegotiator serverTls(final SslContext sslContext) {
+  public static ProtocolNegotiator serverTls(final SslContext sslContext,
+      final ObjectPool<? extends Executor> executorPool) {
     Preconditions.checkNotNull(sslContext, "sslContext");
+    final Executor executor;
+    if (executorPool != null) {
+      executor = executorPool.getObject();
+    } else {
+      executor = null;
+    }
     return new ProtocolNegotiator() {
       @Override
       public ChannelHandler newHandler(GrpcHttp2ConnectionHandler handler) {
         ChannelHandler gnh = new GrpcNegotiationHandler(handler);
-        ChannelHandler sth = new ServerTlsHandler(gnh, sslContext);
+        // The handlers here can out-live the {@link ProtocolNegotiator}.
+        // To keep their own reference to executor from executorPool, we use an extra (unused)
```

Comment:
This comment looks like it was intended for line 118 instead of here. There isn't actually an executor reference here, and it seems to imply the one in the ServerTlsHandler constructor, but that one is not unused.

Reply:
Thanks for the catch! The comments are updated. Since I don't have the write access, could you also merge this PR if everything looks fine? Thanks in advance!